JsonParser.cpp

```diff
@@ -38,5 +49,70 @@ std::map<std::string, std::string> JsonParser::Parse(std::ifstream &fileStream)
 std::map<std::string, std::string> JsonParser::Parse(std::string fileName)
 {
     std::ifstream fstream("units/" + fileName);
-    return JsonParser::Parse(fstream);
+    if (!fstream.good())
+        throw "problem with file stream";
+
+    std::string s((std::istreambuf_iterator<char>(fstream)),
+                  std::istreambuf_iterator<char>());
+    JsonParser::CheckJsonIntegrity(s);
+
+    fstream.clear();
+    fstream.seekg(0);
+
+    std::map<std::string, std::string> myMap;
+    for (std::string::size_type i = 0; i < s.size(); i++)
+    {
+        if (s[i] == '\"')
+        {
+            std::string key = "";
+            for (i++; s[i] != '\"'; i++)
+                key += s[i];
+
+            for (i++; s[i] == ':' || s[i] == ' ' || s[i] == '\"'; i++) // skip until val starts
+                ;
+
+            std::string val = "";
+            for (; s[i] != '\"' && s[i] != '\n' && s[i] != ','; i++)
+                val += s[i];
+
+            myMap[key] = val;
+        }
+    }
+
+    return myMap;
+}
+
+void JsonParser::CheckJsonIntegrity(std::string jsonStr)
+{
+    int numOfSymbols = 5;
+    char symbols[numOfSymbols] = {'{', '}', '\"', ':', ','};
+
+    std::map<char, int> symbolCount;
+    symbolCount['{'] = 0;
+    symbolCount['}'] = 0;
+    symbolCount['\"'] = 0;
+    symbolCount[':'] = 0;
+    symbolCount[','] = 0;
+
+    int expectedCount[numOfSymbols] = {1, 1, 8, 3, 2};
```

Comment:
Ezzel meg az a baj, hogy pl. az idézőjelek száma nagyon függ a felhasználandó kulcsok és string típusú értékek számától. Vagyis, ha Máté kér egy újabb bővítést a programban, akkor ezt a korábban már elkészített részt is újra módosítanod kell majd, vagy legalább végig kell gondolnod, szükséges-e módosítani. Hosszú távon ez sok munkát és potenciális hibát rejt magában.